src/components/NewProjectFlow/index.tsx

```diff
@@ -0,0 +1,184 @@
+import SelectBatches from './flow/SelectBatches';
+import SelectPlantingSites from './flow/SelectPlantingSites';
+import { getSaveText } from './flow/util';
+
+export type FlowStates = 'label' | 'accessions' | 'batches' | 'plantingSites';
+const STATE_ORDER: FlowStates[] = ['label', 'accessions', 'batches', 'plantingSites'];
+
+type NewProjectFlowProps = {
+  reloadData: () => void;
+};
+
+export default function NewProjectFlow({ reloadData }: NewProjectFlowProps): JSX.Element {
+  const { selectedOrganization } = useOrganization();
+  const snackbar = useSnackbar();
+  const history = useHistory();
+
+  const [record, setRecord] = useForm<CreateProjectRequest>({
+    name: '',
+    organizationId: selectedOrganization.id,
+  });
+
+  const [flowState, setFlowState] = useState<FlowStates>('label');
+  const [hasAccessions, setHasAccessions] = useState<boolean>(true);
+  const [hasBatches, setHasBatches] = useState<boolean>(true);
+  const [hasPlantingSites, setHasPlantingSites] = useState<boolean>(true);
+  const [saveText, setSaveText] = useState('');
+  const [projectAccessions, setProjectAccessions] = useState<SearchResponseAccession[]>([]);
+  const [projectBatches, setProjectBatches] = useState<SearchResponseBatches[]>([]);
+  const [projectPlantingSites, setProjectPlantingSites] = useState<PlantingSiteSearchResult[]>([]);
+
+  const goToProjects = useCallback(() => {
+    history.push({ pathname: APP_PATHS.PROJECTS });
+  }, [history]);
+
+  const saveProject = useCallback(async () => {
+    // first create the project
+    let projectId = -1;
+    const response = await ProjectsService.createProject(record);
+    if (!response.requestSucceeded) {
+      snackbar.toastError();
+      return;
+    } else {
+      projectId = response.data.id;
+    }
+
+    if (!projectId) {
+      snackbar.toastError();
+      return;
+    }
+
+    await ProjectsService.assignProjectToEntities(projectId, {
+      accessionIds: projectAccessions.map((entity) => Number(entity.id)),
+      batchIds: projectBatches.map((entity) => Number(entity.id)),
+      plantingSiteIds: projectPlantingSites.map((entity) => Number(entity.id)),
+    });
+
+    reloadData();
+
+    // set snackbar with status
+    snackbar.toastSuccess(
+      getFormattedSuccessMessages(record.name, projectAccessions, projectBatches, projectPlantingSites),
+      strings.formatString(strings.PROJECT_ADDED, record.name) as string
+    );
+
+    // redirect to projects
+    goToProjects();
+  }, [goToProjects, projectAccessions, projectBatches, projectPlantingSites, record, reloadData, snackbar]);
+
+  // If we get to the end of the flowState (there are no more valid steps available), save the project and redirect
+  useEffect(() => {
+    if (!flowState) {
+      void saveProject();
+    }
+  }, [flowState, saveProject]);
+
+  useEffect(() => {
+    setSaveText(getSaveText(flowState, hasAccessions, hasBatches, hasPlantingSites));
```

Comment:
probably want this changing on the activeLocale in useLocalization as well .. same for other usages